### DISCUSSION TITLE
Rush Rebuild Problem

### DISCUSSION THREAD

**Discussion by DeepakMalkan** [iTwin/itwinjs-core#1827] [Q&A] [answered]
'Rush Rebuild' in iModelJS top-level folder seems to run into a problem at the following stage:

=====================

_==[ ninezone-test-app ]==========================================[ 70 of 89 ]==
"ninezone-test-app" had an empty script.

==[ @bentley/extension-cli ]=====================================[ 71 of 89 ]==
"@bentley/extension-cli" completed successfully in 6.06 seconds.

==[ imodel-transformer ]=========================================[ 72 of 89 ]==
"imodel-transformer" completed successfully in 6.49 seconds.

==[ example-code-app ]===========================================[ 73 of 89 ]==
"example-code-app" completed successfully in 7.79 seconds.

==[ @bentley/presentation-frontend ]=============================[ 74 of 89 ]==
"@bentley/presentation-frontend" completed successfully in 15.42 seconds.

==[ core-full-stack-tests ]======================================[ 75 of 89 ]==
One CLI for webpack must be installed. These are recommended choices, delivered as separate packages:
 - webpack-cli (https://github.com/webpack/webpack-cli)
   The original webpack full-featured CLI.
We will use "npm" to install the CLI via "npm install -D".
Do you want to install 'webpack-cli' (yes/no): ^CTerminate batch job (Y/N)? y_

=========================

At the point when it asks to install 'webpack-cli (yes/no):' --- it doesn't take any input characters. Only thing I can do is to terminate the batch job.

Questions:
1. Is 'Rush Rebuild' a valid command to invoke? The package.json file does not mention the rebuild command, though Rush documentation recommends it to completely rebuild all packages from scratch.
2. If 'Rush Clean' followed by 'Rush Build' the equivalent of 'Rush Rebuild'?

I guess I could just install webpack independently.

Deepak.

**Comment by calebmshafer** [comment]
1. Yes, the `rush rebuild` command is valid to run and it basically force re-runs all `npm run build` scripts within the repository. The difference between that and `rush build` is that Rush will only build things that haven't previously been run without a change to it.  Whereas `rush rebuild` is the 
2. It's similar but not quite the same. `rush rebuild` only reruns the actual `npm run build` script in each package and a combintation of `rush clean && rush build` runs both the `npm run clean` and `npm run build` on each package.

That all being said, I'm a bit lost on the webpack-cli issue that they're running into. What Node version are you using?

**Reply by DeepakMalkan** [reply]
Node version: 14.17.0.

I decided not to install webpack separately, since I now think it should be a local install rather than a global one.

**Reply by aruniverse** [reply]
> Node version: 14.17.0.
> 
> 
> 
> I decided not to install webpack separately, since I now think it should be a local install rather than a global one.

Could you try Node 12.17 and higher in 12.x? I don't believe we officially support 14 yet

**Reply by kabentley** [reply]
I use Node 14.16.1 with no problems. Are you sure you did a `rush install` after pulling changes?

**Reply by calebmshafer** [reply]
@DeepakMalkan does that mean you're no longer seeing the issue or does it still happen?

**Reply by DeepakMalkan** [reply]
I've been off-development this week. Get back to it next & revert to this issue then.

**Reply by shenshengyi** [reply]
Hi, I encountered the same problem in the previous version of rush build, but rush build succeeded in the latest imodel.js version. So, you can update the latest code.

**Comment by wgoehrig** [comment, marked as answer]
For anyone still hitting this, `rush purge && rush install` should fix the issue.
